core/src/task.cpp

```diff
@@ -53,11 +53,6 @@ namespace moveit { namespace task_constructor {
 Task::Task(const std::string& id, ContainerBase::pointer &&container)
    : WrapperBase(std::string(), std::move(container)), id_(id)
 {
-	robot_model_loader::RobotModelLoader rml;
-	robot_model_ = rml.getModel();
-	if (!robot_model_)
-		throw Exception("Task failed to construct RobotModel");
-
```

Comment:
I wanted the user to decide how to configure the tasks' robot model. I'm still supporting to invariant that the task provides the robot model. However, only after you explicitly configured it.
Particularly, I wanted to support the use case, that a RobotModel is already available by the application and can be reused...
If you insist, one could auto-load the robot model, when accessing it via `getRobotModel()`.

Reply:
> Particularly, I wanted to support the use case, that a RobotModel is already available by the application and can be reused...

It makes full sense to support this case. Do you insist on setters for this?
Otherwise I would just propose a new (set of?) constructors that can use a `RobotModelPtr` instead.
I think "lazy" loading is a bad idea.

Reply:
Yeah, setters would be better.